source/read/retrieve.txt

```diff
@@ -0,0 +1,227 @@
+
+The following examples use the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Spanish"``. Select the corresponding tab to see
+an example in the Python shell or a ``.py`` file. 
+
+.. tabs::
+
+   .. tab:: Shell
+      :tabid: find-multiple-shell
+
+      .. code-block:: python
+         :copyable: true
+
+         >>> sample_restaurants.restaurants.find({"cuisine": "Spanish"})
+
+   .. tab:: File
+      :tabid: find-multiple-file
+
+      .. code-block:: python
+         :copyable: true
+
+         cursor = sample_restaurants.restaurants.find({"hello": "Spanish"})
+
+You can iterate over the documents in a cursor by using a ``for-in`` loop, as shown in
+the following examples:
+
+.. tabs::
+
+   .. tab:: Shell
+      :tabid: iterate-shell
+
+      .. code-block:: python
+         :copyable: true
+
+         >>> for restaurant in sample_restaurants.restaurants.find({"cuisine": "Spanish"}):
+
+   .. tab:: File
+      :tabid: iterate-file
+
+      .. code-block:: python
+         :copyable: true
+
+         cursor = sample_restaurants.restaurants.find({"hello": "Spanish"})
+         for restaurant in cursor:
+
+.. TODO: To see a full example of using the ``Find()`` method to find multiple documents,
+   see :ref:`pymongo-retrieve-additional-information`.
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass an empty filter 
+   to the ``find()`` method:
+
+   .. code-block:: python
+
+      all_restaurants = sample_restaurants.restaurants.find()
+
+.. TODO:   To see a fully runnable example of using the ``find()`` method to find all documents, see 
+   :ref:`pymongo-retrieve-additional-information`.   
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` and ``find_one()`` methods by passing
+named arguments to them. The following table describes the commonly used arguments:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Argument
+     - Description
+
+   * - ``batch_size`` 
+     - | Limits the number of documents to hold in a cursor at a given time.
+
+   * - ``collation`` 
+     - | An instance of the ``Collation`` class that sets the collation options. 
+
+   * - ``comment`` 
+     - | A string to attach to the query. This can help you trace and interpret the
+         operation in the server logs and in profile data. To learn more about query comments, 
+         see the :manual:`$comment </reference/operator/query/comment/>` page.
+   
+   * - ``hint`` 
+     - | The index to use for the query.
+
+   * - ``max_time_ms`` 
+     - | The maximum execution time on the server for this operation. If this time is
+         exceeded, {+driver-short+} aborts the operation and raises an ``ExecutionTimeout``.
+
+For a full list of available arguments, see the
+`API documentation <{+api-root+}pymongo/collection.html#pymongo.collection.Collection.find>`__
+for the  ``find() method``.
+
+.. TODO: .. instruqt lab
```

Comment:
Is this Instruqt lab for the Modify Find Behavior section?

Reply:
No, I think it will be just for a Find.

Reply:
Okay makes sense. I think we might also want to add a TODO to add some code examples here of using the options in the "Modify" table. Unless that's going to go on one of the more in depth pages?